### DISCUSSION TITLE
How can I access admin UI in Production environment

### DISCUSSION THREAD

**Discussion by fcrios145** [pocketbase/pocketbase#5071] [Q&A]
```go
app.OnBeforeServe().Add(func(e *core.ServeEvent) error {
    e.Router.GET("/*", echo.StaticDirectoryHandler(os.DirFS("./pb_public/browser"), false), middleware.Gzip())
    return nil
})
```
I have that piece of code in my main.go file

* I'm using pocketbase as a framework
* I'm serving my app as standalone binary _no docker, no reverse proxy_
* I'm using angular as my frontend and I'm building the frontend project under pb_public/browser

I'm serving my app through this command.

`./pb serve myapp.example.com` 
Here I'm using a real domain name, and overall it works, 

Api calls, and all, but Whenever I tried to log into the admin page, AKA, https://myapp.example.com/_/
I keep getting the 404 error page from my angular app seems like I'm getting redirected.
Do you know how can I avoid that, so I can access my admin page

**Comment by ganigeorgiev** [comment]
I don't think your issue is PocketBase related, especially if you are not able to reproduce it locally.

Also note that for serving static files I recommend using [`apis.StaticDirectoryHandler`](https://pocketbase.io/docs/go-routing/#serving-static-files) and not `echo.StaticDirectoryHandler` (while they share the same name their implementation slightly differ and the second argument of `apis.StaticDirectoryHandler` allows `index.html` fallbacks; _this confusion is eliminated with the [ongoing refactoring](https://github.com/pocketbase/pocketbase/discussions/4355)_).

If you still think that your problem is PocketBase related, feel free to provide a **minimal** repo or at least a more complete code sample and I'll try to have a more detailed look at it.

---
_edit: It is also possible that the redirect issue you are experiencing is maybe because we have a global `RemoveTrailingSlash` middleware registered by default and conflicts with the echo handler. If that's the case, replacing it with `apis.StaticDirectoryHandler` should fix it (again note that the last argument differs). More details about the method you can find in https://pkg.go.dev/github.com/pocketbase/pocketbase@v0.22.13/apis#StaticDirectoryHandler._
